src/components/tabs/tabs.vue

```diff
@@ -0,0 +1,184 @@
+<template>
+    <div class="ui-tabs">
+        <ul class="ui-tabs--list">
+            <li class="ui-tabs--list-item" v-for="(tab, index) in tabs" :key="index">
+                <a :class="tabClasses(tab)"
+                        :href="tab.href"
+                        class="ui-tabs--link"
+                        tabindex="0"
+                        :target="tab.target"
+                        :rel="tab.rel"
+                        v-if="tab.href !== ''"
+                        ref="tabRefs"
+                        @click.prevent="(e) => activateTab(e, tab)"
+                        @keypress.enter="(e) => activateTab(e, tab)"
+                        @transitionend="onTransitionEnd"
+                        @transitionstart="onTransitionStart">
+                    <render-label :label="tab.$slots['tab-label']" v-if="tab.$slots['tab-label']"/>
+                </a>
+                <div :class="tabClasses(tab)"
+                        :href="tab.href"
+                        class="ui-tabs--link"
+                        tabindex="0"
+                        v-else
+                        ref="tabRefs"
+                        @click.prevent="(e) => activateTab(e, tab)"
+                        @keypress.enter="(e) => activateTab(e, tab)"
+                        @transitionend="onTransitionEnd"
+                        @transitionstart="onTransitionStart">
+                    <render-label :label="tab.$slots['tab-label']" v-if="tab.$slots['tab-label']"/>
+                </div>
+            </li>
+        </ul>
+        <div class="ui-tabs--body" ref="content">
+            <slot/>
+        </div>
+    </div>
+</template>
+
+<script>
+    import { bemMixin } from '../../mixins';
+    
+    export default {
+        components: {
+            //If we want to render not just strings we need to render The label as a Vue Component
+            RenderLabel: {
+                functional: true,
+                render: (h, ctx) => ctx.props.label,
+            },
+        },
+        mixins: [
+            bemMixin('ui-tabs'),
+        ],
+        props: {
+            selectedTabId: {
+                type: Number || null,
+                default: null,
+            },
+            hasAnimation: {
+                type: Boolean,
+                default: true,
+            },
+            shouldAlwaysRenderContent: {
+                type: Boolean,
+                default: false,
+            },
+        },
+        data() {
+            return {
+                initTabs: [],
+                tabs: [],
+                state: {
+                    // We need the corresponding tab from tabs to set the Body active
```

Comment:
Nested Objekte sind immer gefährlich. Ich bin nicht sicher ob alle die States hier nötig sind. Wir sollen dieses Feature mit [Vue.js `<transition>`](https://vuejs.org/v2/guide/transitions.html) umsetzen.

Reply:
Warum sind nested Objekte gefährlich?

Reply:
- Flat [ist fast immer besser als Nested](https://zen-of-python.info/flat-is-better-than-nested.html) es hat viele artikel über das Thema. In diesem spezifichen Fall `state` ist unnötig (wir haben es zusammen mit [meinem Beispiel gesehen](https://codesandbox.io/s/reverent-easley-1wjnx?file=/src/components/Tabs.vue))
- Nested Javascript objekte sind [schwieriger zu handeln](https://medium.com/better-programming/4-ways-to-safely-access-nested-objects-in-vanilla-javascript-8671d09348a8) und es ist enfacher mit den unbewussten Bugs in deinem Code hinzuzufügen
- Vue.js benutzt [javascript Proxies](https://developer.mozilla.org/it/docs/Web/JavaScript/Reference/Global_Objects/Proxy) um die Properties reactive zu machen. Mit Nested Objects ist einfacher dass deine DOM Reactive Expressions (oder auch deine Watchers Callbacks) mehr mals aufgerufen werden bei den tiefen Nested Objekte Attributen.